smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/providers/connectors/WorkerPoolRegistry.java

```diff
@@ -74,6 +116,9 @@ public void init() {
     }
 
     public <T> Uni<T> executeWork(Context msgContext, Uni<T> uni, String workerName, boolean ordered) {
+        if (closed) {
+            throw new IllegalStateException("WorkerPoolRegistry is closed");
```

Comment:
Should it be a rejected operation exception, to follow the executor semantic?

Reply:
I don't have a strong opinion on this. I've considered rejected operation exception, but it is typically thrown by `Executor`s. We also liberally use `UnsupportedOperationException`, while it is part of the Java Collections Framework.